apis/Google.Bigquery.V2/Google.Bigquery.V2/BigqueryParameter.cs

```diff
@@ -0,0 +1,435 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Apis.Bigquery.v2.Data;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using static System.Globalization.CultureInfo;
+
+namespace Google.Bigquery.V2
+{
+    // TODO: Support struct parameters.
+
+    /// <summary>
+    /// A parameter within the <see cref="BigqueryParameterCollection"/> of a <see cref="BigqueryCommand"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Each parameter has a name, type and value. Names are irrelevant for queries using
+    /// positional parameters. All scalar types are converted into strings
+    /// before being passed to the server, but the nature of the conversion depends on the specified type.
+    /// </para>
+    /// <para>
+    /// The following list shows the correspondence between the parameter type and valid value types.
+    /// All scalar types may be specified as a string value, which will not be validated or converted on the client,
+    /// but passed to the API as-is.
+    /// </para>
+    /// <list type="bullet">
+    ///   <item><description><c>Bool</c>: <c>System.Boolean</c></description></item>
+    ///   <item><description><c>Int64</c>: <c>System.Int16</c>, <c>System.Int32</c>, <c>System.Int64</c>, <c>System.UInt16</c>, <c>System.UInt32</c>,
+    ///   <c>System.UInt64</c></description></item>
+    ///   <item><description><c>Float64</c>: Any type valid for <c>Int64</c>, as well as <c>System.Single</c> and <c>System.Double</c></description></item>
+    ///   <item><description><c>String</c>: <c>System.String</c></description></item>
```

Comment:
While we *could* do so, I'd rather not until it's requested. It's easy enough to just call ToString on it from the client code.